### PR TITLE
Fixenvvar

### PR DESCRIPTION
**Pull request by smile21prc**
Use HOME path from env var instead of hard coding it.

### CODE REVIEW

**Comment by smile21prc** [comment]
@MattGal @chcosta , could you please take a look at this PR? Thanks.

**Comment by smile21prc** [comment]
@MattGal, Thank Matt for the review. This whole clean-up step isn't added for OSX, so not need to test this change for OSX now:
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=6386

And I agree with you, the whole script needs to be tested for OSX when it's enabled.

File: src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs [hunk 1]
```diff
@@ -414,8 +414,8 @@ private string GetTEMPDirectory()
                     return Environment.GetEnvironmentVariable("TMPDIR");
                 else if (DirExists(Environment.GetEnvironmentVariable("TMP")))
                     return Environment.GetEnvironmentVariable("TMP");
-                else if (DirExists("/home/DDITAdministrator/myagent/_work/_temp"))
-                    return "/home/DDITAdministrator/myagent/_work/_temp";
+                else if (DirExists(Path.Combine(Environment.GetEnvironmentVariable("HOME"), "myagent/_work/_temp")))
```
Comment (chcosta) [inline src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]: If all you're doing is checking for the existence of a directory as your "temp" directory, why does it matter what OS you're on?

Comment (chcosta) [inline src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]:
```C#
private void AddDirToListIfExist(List<string> dirs, string directory)
```
`dirs` could be a HashSet

Comment (chcosta) [inline src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]: Why not follow the `AddDirToListIfExist` pattern which is used in GetNugetCacheDirectories?

Comment (chcosta) [inline src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]: Method name should be `AddDirToListIfExists`, missing trailing 's'

**Pull request merged**